File: semester_02/databases/src/01_connection/program-conn.py
import os
import sys
import psycopg2
from psycopg2 import sql
from dotenv import load_dotenv

load_dotenv()
HOST     = os.getenv("PG_HOST")
PORT     = os.getenv("PG_PORT", "5432")
DBNAME   = os.getenv("PG_DBNAME")
USER     = os.getenv("PG_USER")
PASSWORD = os.getenv("PG_PASSWORD")
TABLE    = os.getenv("PG_TABLE")

if not all([HOST, DBNAME, USER, PASSWORD, TABLE]):
    sys.exit("Uzupelnij plik .env (PG_HOST, PG_DBNAME, PG_USER, PG_PASSWORD, PG_TABLE)")

# Polaczenie z baza
conn = psycopg2.connect(
    host=HOST, port=int(PORT), dbname=DBNAME,
    user=USER, password=PASSWORD, gssencmode="disable"
)
print(f"Polaczono z baza '{DBNAME}' na {HOST}\n")
cur = conn.cursor()

# Pobieranie nazwy kolumn z bazy
cur.execute("""
    SELECT column_name
    FROM information_schema.columns
    WHERE table_name = %s AND table_schema = 'public'
    ORDER BY ordinal_position
""", (TABLE,))

columns = [row[0] for row in cur.fetchall()]
# Pobranie danych z tabeli
cur.execute(sql.SQL("SELECT * FROM {}").format(sql.Identifier(TABLE)))
rows = cur.fetchall()

# Wyswietlenie danych
print(" | ".join(columns))
print("-" * 60)
for row in rows:
    print(" | ".join(str(cell) for cell in row))
print(f"\nLiczba wierszy: {len(rows)}")

cur.close()
conn.close()

File: semester_02/databases/src/01_connection/program-conn.cs
using System;
using System.Collections.Generic;
using System.IO;
using Microsoft.Data.SqlClient;

// Wczytanie danych logowania z pliku appsettings.conf
var config = new Dictionary<string, string>();
foreach (var line in File.ReadAllLines("appsettings.conf"))
{
    if (string.IsNullOrWhiteSpace(line) || line.StartsWith("#")) continue;
    var parts = line.Split('=', 2);
    if (parts.Length == 2)
        config[parts[0].Trim()] = parts[1].Trim();
}

string server = config["MSSQL_SERVER"];
string database = config["MSSQL_DATABASE"];
string user = config["MSSQL_USER"];
string password = config["MSSQL_PASSWORD"];
string table = config["MSSQL_TABLE"];

// Polaczenie z baza
string connStr = $"Server={server};Database={database};User Id={user};Password={password};TrustServerCertificate=True;Integrated Security=False;";
using var conn = new SqlConnection(connStr);
conn.Open();
Console.WriteLine($"Polaczono z baza '{database}' na {server}\n");

// Pobranie nazwy kolumn z bazy
var columns = new List<string>();
using (var cmd = new SqlCommand(
    "SELECT COLUMN_NAME FROM INFORMATION_SCHEMA.COLUMNS " +
    "WHERE TABLE_NAME = @t ORDER BY ORDINAL_POSITION", conn))
{
    cmd.Parameters.AddWithValue("@t", table);
    using var reader = cmd.ExecuteReader();
    while (reader.Read())
        columns.Add(reader.GetString(0));
}

Console.WriteLine($"Kolumny tabeli '{table}': {string.Join(", ", columns)}\n");

// Pobranie i wyświetlenie dane
using var cmd2 = new SqlCommand($"SELECT * FROM [{table}]", conn);
using var data = cmd2.ExecuteReader();

Console.WriteLine(string.Join(" | ", columns));
Console.WriteLine(new string('-', 60));

int count = 0;
while (data.Read())
{
    var cells = new List<string>();
    for (int i = 0; i < data.FieldCount; i++)
        cells.Add(data.IsDBNull(i) ? "NULL" : data.GetValue(i).ToString()!);
    Console.WriteLine(string.Join(" | ", cells));
    count++;
}

Console.WriteLine($"\nLiczba wierszy: {count}");